tools/relay-cache-warmer/graphql.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const query = `query TwinsQuery {
+  items: twinsConnection(first: %d, orderBy: id_ASC%s) {
+    pageInfo {
+  	  endCursor
+  	  hasNextPage
+    }
+  }
+  twins(limit: %d, offset: %s, orderBy: id_ASC) {
+    accountID
+    publicKey
+    twinID
+    relay
+  }
+}`
+
+const twinsPerPage = 500
+
+type Twin struct {
+	ID        uint      `json:"id"`
+	AccountID string    `json:"account"`
+	Relay     []string  `json:"relay"`
+	PK        JSONArray `json:"pk"`
+}
+
+// only exists because json marshals []byte to a string instead of json array
+type JSONArray []byte
+
+func (u JSONArray) MarshalJSON() ([]byte, error) {
+	if u == nil || len(u) == 0 {
+		return []byte("null"), nil
+	}
+
+	var buf bytes.Buffer
+	_, err := buf.WriteRune('[')
+	if err != nil {
+		return nil, err
+	}
+	for i, b := range u {
+		if i != 0 {
+			_, err = buf.WriteRune(',')
+			if err != nil {
+				return nil, err
+			}
+		}
+		_, err = buf.WriteString(strconv.FormatUint(uint64(b), 10))
+		if err != nil {
+			return nil, err
+		}
+	}
+	_, err = buf.WriteRune(']')
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+
+}
+
+func getTwins(graphql string) ([]Twin, error) {
```

Comment:
The point of using pagination is that you can process the data in batches. It is not right to use pagination to still collect ALL twins in one list !! then save them all

Instead this should work as:
- get first batch (say 200 twin)
- save 200 twin
- take a very short break 
- get next 200
- save 
- and so on!